Source/ASTextNode2.mm

```diff
@@ -78,6 +70,12 @@ @implementation ASTextNode2 {
   NSAttributedString *_composedTruncationText;
   NSArray<NSNumber *> *_pointSizeScaleFactors;
   
+  // If nil, regenerate.
+  NSAttributedString *_preparedAttributedText;
+  
+  // Just a fast cache. May not be valid but good place to check first.
```

Comment:
Could you clarify this comment a bit, specifically what about it may not be valid (like if constrainedSize doesn't match?)

Same for "If nil, regenerate" -- could remove this comment, or make it a bit more clear how the variable is used and expectations for how it is maintained.